### PR TITLE
theme designer: add accessibility checker pairs

### PR DESCRIPTION
**Pull request by emmayjiang**
Adds more pairs to the accessibility checker (and removes the one that was obviously wrong).

image
image
image

### CODE REVIEW

File: packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx [hunk 1]
```diff
@@ -53,9 +53,194 @@ export const AccessibilityChecker: React.FunctionComponent<AccessibilityCheckerP
       'NeutralForeground1 on NeutralBackground1',
     );
     calculateContrastRatio(
-      input.colorNeutralForeground1,
-      input.colorNeutralForeground2,
-      'NeutralBackground1 on NeutralBackground2',
+      input.colorNeutralForeground2BrandHover,
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx:56]: turn this into data that you loop over. and remove the manual listing of key name. You should be able to pull that from the data.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/82337ca348673485e78b9072431c00c222cd985a'>82337ca348673485e78b9072431c00c222cd985a</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx [hunk 2]
```diff
@@ -21,21 +21,23 @@ export const AccessibilityChecker: React.FunctionComponent<AccessibilityCheckerP
...
-    const bgc: Vec3 = hex_to_sRGB(background);
-    const fgc: Vec3 = hex_to_sRGB(foreground);
+  const calculateContrastRatio = (foreground: string, background: string) => {
+    const theme = (props.theme as unknown) as Record<string, string>;
 
-    const currContrastRatio = getContrastRatio(bgc, fgc);
-    const contrastRatioString = currContrastRatio.toFixed(2);
+    const bghex: string = theme[foreground];
+    const fghex: string = theme[background];
+
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx:29]: we're pretty confident that bgHex and fgHex will be returned, but lets guard against the case where they don't return a value

if (!bghex || !fghex) return;

File: packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx [hunk 3]
```diff
@@ -21,21 +21,23 @@ export const AccessibilityChecker: React.FunctionComponent<AccessibilityCheckerP
...
-  const calculateContrastRatio = (foreground: string, background: string, colorPairString: string) => {
-    const bgc: Vec3 = hex_to_sRGB(background);
-    const fgc: Vec3 = hex_to_sRGB(foreground);
+  const calculateContrastRatio = (foreground: string, background: string) => {
+    const theme = (props.theme as unknown) as Record<string, string>;
 
-    const currContrastRatio = getContrastRatio(bgc, fgc);
-    const contrastRatioString = currContrastRatio.toFixed(2);
+    const bghex: string = theme[foreground];
+    const fghex: string = theme[background];
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx:28]: camelCase

```suggestion
    const bgHex: string = theme[foreground];
    const fgHex: string = theme[background];
```

File: packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx [hunk 4]
```diff
@@ -21,21 +21,23 @@ export const AccessibilityChecker: React.FunctionComponent<AccessibilityCheckerP
...
+
+    const bgc: Vec3 = hex_to_sRGB(bghex);
+    const fgc: Vec3 = hex_to_sRGB(fghex);
 
-    const currContrastRatioPair = background + ' on ' + foreground;
+    const currContrastRatio = getContrastRatio(bgc, fgc);
```
[the remark this answers is not in the document]
Comment (emmayjiang) [reply packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx:33]: it's used again later

[the remark this answers is not in the document]
Comment (emmayjiang) [reply packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx:33]: if (currContrastRatio < 4.5) {

**Review by micahgodbolt — APPROVED** [review]

**Review by Retracted81 — APPROVED** [review]

File: packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx (hunk 1 above)
Comment (Retracted81) [reply packages/react-components/theme-designer/src/components/AccessibilityChecker/AccessibilityChecker.tsx]: Im new at this been blocked by crooks someone was using my old device to code help would be appreciated been 6 months of lies from everyone please understand its hard to trust anyone

**Pull request merged**